lib/resourcebuilder/apps.go

```diff
@@ -77,6 +81,21 @@ func waitForDeploymentCompletion(ctx context.Context, client appsclientv1.Deploy
 		klog.V(4).Infof("Deployment %s is not ready. status: (replicas: %d, updated: %d, ready: %d, unavailable: %d)", d.Name, d.Status.Replicas, d.Status.UpdatedReplicas, d.Status.ReadyReplicas, d.Status.UnavailableReplicas)
 		return false, nil
 	}, ctx.Done())
+
+	if err == wait.ErrWaitTimeout {
+		message := fmt.Sprintf("Deployment %s failed to become ready in time.", d.Name)
```

Comment:
We don't use terminology that indicates "past" - we would say "Deployment is not yet available: ..."